cmd/main.go

```diff
@@ -35,34 +33,5 @@ func main() {
 		}()
 	}
 
-	panic(startServer(config, logger))
-}
-
-func startServer(config *config.Config, logger *logrus.Logger) (err error) {
-	logger.Infoln("starting")
-	mux := http.NewServeMux()
-	registry := streaming.NewSocketRegistry()
-
-	monitoring.StartProfilerServer(config, mux, logger)
-	if config.StatusPort > 0 {
-		monitoring.StartStatusServer(config, logger)
-	}
-	if config.Monitoring != nil {
-		monitoring.StartServerMetrics(config, logger, registry)
-	}
-
-	producerRules, err := config.ConfigureProducers(logger)
-	if err != nil {
-		return err
-	}
-
-	server, _, err := streaming.InitServer(config, mux, producerRules, logger, registry)
-	if err != nil {
-		return err
-	}
-
-	if server.TLSConfig, err = config.ExtractServiceTLSConfig(); err != nil {
-		return err
-	}
-	return server.ListenAndServeTLS(config.TLS.ServerCert, config.TLS.ServerKey)
+	panic(run.RunServer(context.Background(), config, logger))
```

Comment:
hmm.. Would like to keep the method name as `StartServer` as `run.RunServer` almost looks like stuttering